### DISCUSSION TITLE
Edge classification across the whole graph

### DISCUSSION THREAD

**Discussion by lpolisi** [pyg-team/pytorch_geometric#7930] [Q&A]
I have a dataset with a large number of small graphs where each graph contains both node and edge attributes. The edges are fixed, ie no missing link.

For each graph, I want to identify one and only one edge with the label 1, and the rest should be labeled as 0. In all the edge classification examples I have found, the classification is done for each edge individually.

My current idea is to simply do an edge classification on each edge individually and then renormalize the predictions so they sum to 1, and pick the argmax. Is this a valid approach? Are there other better ways of doing this?

**Comment by rusty1s** [comment]
This sounds like a classic binary classification problem (with potentially imbalanced labels), so I would treat it as such, e.g., compute edge-level probabilities and then train via cross entropy. The `argmax` is then your final prediction.
